prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5467,7 +5496,41 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                 (expr.field(field), stmts)
             }
             PlaceEncoding::ArrayAccess { base, index, array_elem_ty, array_len } => {
-                panic!("ArrayAccess {{ {:?}, {:?}, {:?}, {:?} }}", base, index, array_elem_ty, array_len);
+                let elem_ty_name = if let vir::Type::TypedRef(ref name) = array_elem_ty { name } else { unreachable!() };
+                let lookup_pure_ret = rust_type_name_to_vir_type(elem_ty_name)?;
+                let lookup_pure = self.encoder.encode_builtin_function_use(
+                    BuiltinFunctionKind::ArrayLookupPure {
+                        array_elem_ty: array_elem_ty.clone(),
+                        array_len,
+                        return_ty: lookup_pure_ret.clone(),
+                    }
+                );
+
+                let array_type = base.get_type().clone();
+                let lookup_res: vir::Expr = self.get_auxiliary_local_var("lookup_ret", array_elem_ty).into();
+                let lookup_res_val_int = lookup_res.clone().field(vir::Field::new("val_int", vir::Type::Int));
+                let (encoded_base_expr, mut stmts) = self.postprocess_place_encoding(*base)?;
```

Comment:
I think there is a problem when you have multiple nested `ArrayAccess`, because the `get_auxiliary_local_var` call two lines above is designed to reuse generated variables, but you always want a fresh one. I think you could implement something like a new `ProcedureEncoder::generate_fresh_local_var(..)` for that.